main/src/ca/uwaterloo/flix/api/lsp/provider/SymbolProvider.scala

```diff
@@ -15,29 +15,31 @@
  */
 package ca.uwaterloo.flix.api.lsp.provider
 
-import ca.uwaterloo.flix.api.lsp.{DocumentSymbol, Index, Position, Range, SymbolKind, SymbolTag}
+import ca.uwaterloo.flix.api.lsp.{DocumentSymbol, Position, Range, SymbolKind}
+import ca.uwaterloo.flix.language.ast.TypedAst
 import ca.uwaterloo.flix.language.ast.TypedAst.Root
-import ca.uwaterloo.flix.language.debug.Audience
 
 object SymbolProvider {
 
-  private implicit val audience: Audience = Audience.External
+  def processDocumentSymbols(uri: String)(implicit root: Root): List[DocumentSymbol] = root.enums.filter {
+    case (_, enum) => enum.loc.source.name == uri }.map {
+      case (_, enum) => enumToDocumentSymbol(enum)
+    }.toList
 
-  def processDocumentSymbols(uri: String)(implicit index: Index, root: Root): List[DocumentSymbol] =
-    List(DocumentSymbol("name",
-      Some("description"),
-      SymbolKind.Interface,
-      Range(Position(1, 1), Position(2, 5)),
-      Range(Position(1, 1), Position(2, 5)),
-      List(SymbolTag.Deprecated),
-      List(DocumentSymbol(
-        "parent",
-        Some("parentDescription"),
-        SymbolKind.Interface,
-        Range(Position(0, 4), Position(0, 10)),
-        Range(Position(0, 4), Position(0, 10)),
-        List(),
-        List()
-      ))
-    ))
+  private def enumToDocumentSymbol(enum: TypedAst.Enum): DocumentSymbol = DocumentSymbol(
+    enum.sym.name,
+    Some(enum.doc.lines.mkString(" ")),
+    SymbolKind.Enum,
+    Range(
```

Comment:
Here you can simply use `Range.from`.

Reply:
This first range is manually built since I need to pass the documentation start as begin position of it

Reply:
Oh I see.

But I still think it would be better to just use the enum symbol.

Let us worry about documentation later. (Because it brings some problems: Imagine if you have a big copyright header at the top and then an enum. With the current code, the entire copyright header would be part of the symbol. I think, for not at least, only code should be part of the symbol.)